### PR TITLE
Added new FileClient.StartCopy() parameters

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Review by amnguye — APPROVED** [review]

**Review by jaschrep-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/FileSmbProperties.cs [hunk 1]
```diff
@@ -127,22 +127,11 @@ public override bool Equals(object other)
...
-
-        private static string DateTimeOffSetToString(DateTimeOffset dateTimeOffset)
-            => dateTimeOffset.UtcDateTime.ToString(Constants.File.FileTimeFormat, CultureInfo.InvariantCulture);
     }
-  /// <summary>
-  /// FilesModelFactory provides utilities for mocking.
-  /// </summary>
+
+     /// <summary>
+     /// FilesModelFactory provides utilities for mocking.
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/FileSmbProperties.cs]: nit: update type name

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs [hunk 2]
```diff
@@ -587,17 +618,88 @@ private void SetNameFieldsIfNull()
         /// A <see cref="RequestFailedException"/> will be thrown if
         /// a failure occurs.
         /// </remarks>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "<Pending>")]
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs]: Can we add that EditorBrowsable.Never?

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs [hunk 3]
```diff
@@ -587,17 +618,88 @@ private void SetNameFieldsIfNull()
...
+        public virtual Response<ShareFileCopyInfo> StartCopy(
+            Uri sourceUri,
+            Metadata metadata,
+            CancellationToken cancellationToken) =>
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs]: Does leaving CancellationToken as optional cause compiler ambiguity issues?

**Pull request merged**